python/az/aro/azext_aro/_dynamic_validators.py

```diff
@@ -83,13 +84,29 @@ def get_clients(key, cmd):
     return parts, network_client, auth_client
 
 
+def progress_tracking_factory(msg):
```

Comment:
Idiomatic python means we probably wouldn't name it "factory" as it's a decorator function. Using more imperative language like `display_progress` or `track_progress` is more idiomatic in my opinion. I'd also like to see it documented.